src/roguelike/game_engine/game_manager/game_constructor/game_loader.py

```diff
@@ -0,0 +1,92 @@
+"""
+Contains all classes needed to load game
+"""
+
+import json
+import typing as tp
+
+from roguelike.game_engine.env_manager.env_manager import Environment, Inventory
+from roguelike.game_engine.env_manager.map import MapCoordinates, Map
+from roguelike.game_engine.env_manager.map_objects_storage import Stats, Obstacle, Treasure, PlayerCharacter, MapObject
+from roguelike.game_engine.game_manager.game_processor.game_state import GameState, Mode
+
+
+class GameLoader:
+    """Loads GameState from file"""
+
+    @staticmethod
+    def load_game(path: str) -> GameState:
+        with open(path, encoding="utf-8") as json_file:
+            world_data = json.load(json_file)
+        geomap, world_objects, player = GameLoader._load_world(world_data)
+        return GameState(Mode.MAP, Environment(geomap, world_objects), Inventory([]), player)
+
+    @staticmethod
+    def _load_coordinates(value: tp.List[int]) -> MapCoordinates:
+        if len(value) != 2:
+            raise ValueError("Invalid coordinates json")
+        return MapCoordinates(value[0], value[1])
+
+    @staticmethod
+    def _load_stats(value: tp.Dict[str, float]) -> Stats:
+        if len(value) != 2 or "attack" not in value.keys() or "health" not in value.keys():
+            raise ValueError("Invalid stats json")
+        return Stats(value["health"], value["attack"])
+
+    @staticmethod
+    def _load_obstacle(value: tp.Dict[str, tp.Any]) -> Obstacle:
+        if len(value) != 0:
+            raise ValueError("Invalid obstacle settings json")
+        return Obstacle()
+
+    @staticmethod
+    def _load_treasure(value: tp.Dict[str, tp.Any]) -> Treasure:
+        if len(value) != 2 or "name" not in value.keys() or "stats" not in value.keys() or \
+                not isinstance(value["name"], str):
+            raise ValueError("Invalid treasure settings json")
+        return Treasure(value["name"], GameLoader._load_stats(value["stats"]))
+
+    @staticmethod
+    def _load_player(value: tp.Dict[str, tp.Any]) -> PlayerCharacter:
+        if len(value) != 1 or "stats" not in value.keys():
+            raise ValueError("Invalid player settings json")
+        return PlayerCharacter(GameLoader._load_stats(value["stats"]))
+
+    @staticmethod
+    def _load_map(value: tp.Dict[str, int]) -> Map:
+        if len(value) != 2 or "width" not in value.keys() or "height" not in value.keys():
+            raise ValueError("Invalid map json")
+        return Map(value["width"], value["height"])
+
+    @staticmethod
+    def _load_world(value: tp.Dict[str, tp.Any]) -> tp.Tuple[Map, tp.List[tp.Any], PlayerCharacter]:
```

Comment:
Seems like large function, let's split into simple steps, wdyt?

Reply:
fixed this, thanks